hal/src/rtl872x/i2c_hal.cpp

```diff
@@ -687,8 +733,7 @@ void hal_i2c_enable_dma_mode(hal_i2c_interface_t i2c, bool enable, void* reserve
 uint8_t hal_i2c_reset(hal_i2c_interface_t i2c, uint32_t reserved, void* reserved1) {
     auto instance = CHECK_TRUE_RETURN(I2cClass::getInstance(i2c), 1);
     I2cLock lk(instance);
-    instance->reset();
-    return 0;
+    return instance->reset();
```

Comment:
@XuGuohui Let's change the return type here to `int` and return `system_error_t` codes. + change the `TwoWire::reset` to also propagate the error.

Reply:
That's a breaking change, is that okay?

Reply:
In this case it's fine. Return value is not used in wiring, so we can change the type/values.

Reply:
We still needd to introduce a new hal API to not break the dynalib ABI.